Moodis/Moodis/Feature/Login/LoginViewModel.cs

```diff
@@ -22,6 +24,7 @@ public User Authenticate(string username, string password)
 
             if (userList.Exists(x => x.username == username && x.password == Crypto.CalculateMD5Hash(password)))
             {
+                currentUser = userList.Find(x => x.username == username && x.password == Crypto.CalculateMD5Hash(password));
                 return userList.Find(x => x.username == username && x.password == Crypto.CalculateMD5Hash(password));
```

Comment:
Since you return null on unsuccessful Find, you should refactor this code of block to just `return userList.Find(x => x.username == username && x.password == Crypto.CalculateMD5Hash(password));`. That means basically deleting almost whole if block

Reply:
And change that `x` to `user`

Reply:
I don't even know who wrote this but sure.

Reply:
x is already a user in that instance. Also Andrius wrote most of this code i just assigned current user to the user who authenticated.

Reply:
I somewhat changed it in the latest commit hopefully for the better.

Reply:
> x is already a user in that instance. Also Andrius wrote most of this code i just assigned current user to the user who authenticated.

Yes it is, I was referring to naming